fdbserver/BlobManager.actor.cpp

```diff
@@ -1073,14 +1073,30 @@ static bool handleRangeIsRevoke(Reference<BlobManagerData> bmData, RangeAssignme
 		for (auto& it : currentAssignments) {
 			// ensure range doesn't truncate existing ranges
 			if (it.begin() < assignment.keyRange.begin || it.end() > assignment.keyRange.end) {
-				fmt::print("Assignment [{0} - {1}) truncates range [{2} - {3})\n",
-				           assignment.keyRange.begin.printable(),
-				           assignment.keyRange.end.printable(),
-				           it.begin().printable(),
-				           it.end().printable());
+				// the only case where this is ok is on startup when a BM is revoking old granules after reading
+				// knownBlobRanges and seeing that some are no longer present.
+				auto knownRanges = bmData->knownBlobRanges.intersectingRanges(it.range());
+				bool inKnownBlobRanges = false;
+				for (auto& r : knownRanges) {
+					if (r.value()) {
+						inKnownBlobRanges = true;
+						break;
+					}
+				}
+				bool forcePurging = bmData->isForcePurging(it.range());
+				if (it.cvalue() != UID() || (inKnownBlobRanges && !forcePurging)) {
+					fmt::print("Assignment [{0} - {1}): {2} truncates range [{3} - {4}) ({5}, {6})\n",
```

Comment:
That's a good call. It wouldn't be too bad, because this happens right before an assert, but to debug any of this assert failures in prod, we would want to have a trace event. I'll change it

Reply:
Actually i'm doing a follow-up PR to this anyway so i'll just change it there, since CI is passing